agents/README.md

```diff
@@ -0,0 +1,136 @@
+```
+
+Next we need to deploy a Kubernetes secret which will provide the GCP credentials needed to run TensorBoard:
+
+```bash
+SECRET_FILE_NAME="secret.json"
+SECRET_NAME="gcp-credentials"
+KEY_FILE="/path/to/your/key.json"
+kubectl create -n ${NAMESPACE} secret generic ${SECRET_NAME} \
+  --from-file=${SECRET_FILE_NAME}=${KEY_FILE}
+```
+
+This service account key will need read access to Google Cloud Storage in order to permit TensorBoard to read training log data that it will then display in the TensorBoard UI.
+
+Lastly, this example will deploy jobs into the "rl" namespace so let's make sure that exists with the following:
+
+```bash
+NAMESPACE="rl"
+kubectl create namespace ${NAMESPACE}
+```
+
+##### Ksonnet setup
+
+Ksonnet is a tool to simplify configuration management for kubernetes deployments which extends to making it easier to specify and re-configure distributed neural network training jobs on Kubeflow.
+
+Before we can use it we need to do a little setup of our local machine namely registering a particular kubernetes cluster as the default for Ksonnet and pulling the Kubeflow Ksonnet libraries that are needed in order to use Ksonnet to manage Kubeflow training jobs.
+
+These setup steps are automated and you can run them with the following command:
+
+```bash
+sh setup.sh
+```
+
+### Running the demo
+
+As mentioned above this example demonstrates the training of a reinforcement learning agent in performing a robotic control task. There are two aspects to discuss:
+1. The docker image the encapsulates all that is needed to define the training task, and
+2. The Jupyter Notebook container that is sufficient to view the demonstration notebook and dispatch training jobs that utilize the preceeding.
+
+#### Job container
+
+If you're only interested in running this example and examining the intermediate and final outputs you don't necessarily need to re-build the container that encapsulates the training task. A public version of this container can be found here:
+
+If you look ahead to the demonstration notebook you'll see we specify the address of this container when parameterizing our training job.
+
+Alternatively, especially in the case where you want to make modifications to the example, you may want to re-build the training container image. This can be done by running
+
+```bash
+sh build.sh
+```
+
+Once built the image needs to be made available on a container registry accessible to the kubeflow deployment that will be pulling and running it. If your Kubernetes cluster is running on the Google Cloud Platform then a good pattern is to push your image to a Google Container Registry (GCR) that is either
+1. Owned by the same project as your GKE cluster, or
+2. Public
+
+You can push your recent container build to GCR with the following command:
+
+```bash
+gcloud docker -- push ${IMAGE_TAG}
+```
+
+#### Demonstration notebook
+
+There are two ways you can run the demonstration notebook.
+
+###### Jupyter on host OS
```

Comment:
Do we need instructions for Jupyter on OS and Jupyter in Docker? Isn't Kubeflow only instructions sufficient?

Reply:
I think so but thought other people wanted to be able to run it outside of that. Happy to remove these other branches it would make things a lot clearer.